src/main/java/com/manoelcampos/javadoc/coverage/stats/ClassMembersDocStats.java

```diff
@@ -66,14 +69,34 @@ public long getMembersNumber() {
          * a class source code) will be computed as undocumented.
          */
         return Arrays.stream(membersDocs)
+                .filter(filterPublicIfNecessary())
                 .map(Doc::position)
                 .filter(Objects::nonNull)
                 .count();
     }
 
     @Override
     public long getDocumentedMembers() {
-        return Arrays.stream(membersDocs).map(Doc::getRawCommentText).filter(Utils::isNotStringEmpty).count();
+        return Arrays.stream(membersDocs)
+                .filter(filterPublicIfNecessary())
+                .map(Doc::getRawCommentText)
+                .filter(Utils::isNotStringEmpty)
+                .count();
+    }
+
+    private Predicate<? super Doc> filterPublicIfNecessary() {
```

Comment:
You don't need to make the method return a `Predicate`. That just makes the method body more complex by requiring it to return a lambda expression. You can just change the method signature to the same signature of the `test` method into the `Predicate` functional interface: `private boolean filterPublicIfNecessary(Doc m)`.

Then, where you have `filterPublicIfNecessary()` you just change to a method reference `this::filterPublicIfNecessary`

Reply:
changed